touchforms/backend/test_xformplayer.py

```diff
@@ -75,7 +75,7 @@ def mock_query(query_function, criteria):
             self.xform,
             session_data=self.session_data,
             form_context={
-                'all_case_ids': ['legolas', 'gimli'],
+                'all_case_ids': ['legolas', 'sauron'],
```

Comment:
hmm so if i understand correctly this worked before since we went in order through the `case_ids` and since we had the case for `legolas` in the `case_model` it stopped before checking gimli. now that we sort, gimli came first and broke it since the gimli case was not included in the `form_context` and thus looked for it on HQ which obviously isn't running during the tests. i think that that's fine, especially since this is really only used so we can test other functionality. it'll just make the tests kind of brittle. if we added the `gimli` case to the `form_context` i think it might've worked (https://github.com/dimagi/touchforms/blob/0efe0ccd3030b49d1ab248127fe2c4c7077ff6b5/touchforms/backend/test_touchcare.py#L67).

anyways thanks for fixing and diving into the java code. looked brutal